### PR TITLE
Saffron Finance Strategy

### PR DESCRIPTION
**Pull request by turpintinz**
Changes proposed in this pull request:
-  Add saffron-finance strategy
-  Sum up balances from several LP Tokens
-  Apply multiplier to score as configured

### CODE REVIEW

File: src/strategies/saffron-finance/index.ts [hunk 1]
```diff
@@ -0,0 +1,268 @@
...
+        boostResponses: new Array(),
+        votingScores: new Array()
+      }
+      let queries = addresses.map((address: any) => [
+        token.tokenAddress,
+        "balanceOf",
+        [address]
+      ]);
+      tokenRecord.queries.push(queries);
+      let response = await multicall(network, provider, abi, queries, {blockTag});
```
Comment (bonustrack) [inline src/strategies/saffron-finance/index.ts:171]: The way the strategy is designed is not going to scale well, the result of a strategy is calculated on user browser i recommend to send a maximum of 3 multicall request in a single strategy but this strategy seem to be sending more than 30 multicall.

Comment (turpintinz) [reply src/strategies/saffron-finance/index.ts:171]: Okay, thank you. I can create an implementation with that restriction.

File: src/strategies/saffron-finance/index.ts [hunk 2]
```diff
@@ -0,0 +1,273 @@
...
+    });
+    votingScores.push(...batchScores);
+  });
+
+  // ================ Sum up everything =================
+  let addressVotingScore = addresses.map((address: any, index: number) => {
+    let total = BigNumber.from(0);
+    holdersQueryBatches.forEach((batch: Batch, index: number) => {
+      let votingScore = votingScores[batch.qIdxStart+index];
+      if (votingScore.address === address) {
```
Comment (bonustrack) [inline src/strategies/saffron-finance/index.ts:254]: @turpintinz Just tried the strategy, it fail for me at this line with `TypeError: Cannot read property 'address' of undefined`.

Comment (turpintinz) [reply src/strategies/saffron-finance/index.ts:254]: Yep, me too. I'll be repairing that today, April 9. Thanks for asking.

Comment (turpintinz) [reply src/strategies/saffron-finance/index.ts:299]: > @turpintinz Just tried the strategy, it fail for me at this line with `TypeError: Cannot read property 'address' of undefined`.

This is now repaired.

**Review by bonustrack — APPROVED** [review]
Works well now

**Pull request merged**